src/index.ts

```diff
@@ -148,13 +161,21 @@ function path(template: string, params: ParamMap) {
  * ```
  */
 export function join(part1: string, separator: string, part2: string) {
-  const p1 = part1.endsWith(separator) ? part1.slice(0, -separator.length) : part1;
-  const p2 = part2.startsWith(separator) ? part2.slice(separator.length) : part2;
-  return p1 === '' || p2 === ''
-    ? p1 + p2
-    : p1 + separator + p2;
+  const p1 = part1.endsWith(separator)
+    ? part1.slice(0, -separator.length)
+    : part1;
+  const p2 = part2.startsWith(separator)
+    ? part2.slice(separator.length)
+    : part2;
+  return p1 === "" || p2 === "" ? p1 + p2 : p1 + separator + p2;
 }
 
+/**
+ * Removes Null and undefined :params.
+ * @param params Object consisting of :params
+ *
+ * @returns {Object} Object of :params after removing Null and undefined values
+ */
```

Comment:
This is an internal function that isn't exported so it doesn't need a jsdoc string.